app/views/unattended/provisioning_templates/PXEGrub2/kickstart_default_pxegrub2.erb

```diff
@@ -32,11 +32,12 @@ menuentry '<%= template_name %>' {
 
 <%
 # Workaround for https://bugzilla.redhat.com/show_bug.cgi?id=1811561 and https://bugzilla.redhat.com/show_bug.cgi?id=1842893
-if @host.provision_interface.subnet.httpboot
-  proxy_http_port = @host.provision_interface.subnet.httpboot.httpboot_http_port
-  proxy_https_port = @host.provision_interface.subnet.httpboot.httpboot_https_port
+subnet = @host.provision_interface.subnet || @host.provision_interface.subnet6
+if subnet.httpboot
```

Comment:
Should be be very safe and not assume anything?
```suggestion
if subnet && subnet.httpboot
```

Not sure if it's allowed in templates, but this may be simplified to
```suggestion
if subnet&.httpboot
```

Reply:
the `&.` won't work in safe mode, not because it's not whitelisted but ruby-parser/sexp_processor don't understand this new ruby syntax

Reply:
Thanks rebasing.